examples/tsdf/online.cc

```diff
@@ -156,41 +168,54 @@ void reconstruct(const ZEDNative &zed_native, const L515 &l515,
                  const std::shared_ptr<inference_engine> &segmentation_engine,
                  const std::string &config_file_path) {
   // initialize TSDF
-  auto TSDF = std::make_shared<TSDFSystem>(0.01, 0.06, 4, get_intrinsics(config_file_path));
+  auto TSDF = std::make_shared<TSDFSystem>(0.01, 0.06, 4,
+      get_intrinsics(config_file_path), get_extrinsics(config_file_path));
   SLAM->startup();
 
   ImageRenderer renderer("tsdf", SLAM, TSDF, config_file_path);
 
-  std::thread t([&]() {
+  auto POSE_MANAGER = std::make_shared<pose_manager>();
+
+  std::thread t_slam([&]() {
+    cv::Mat img_left, img_right;
     while (true) {
-      cv::Mat img_left, img_right, img_rgb, img_depth;
+      //cv::Mat img_left, img_right, img_rgb, img_depth;
       if (SLAM->terminate_is_requested())
         break;
       // get sensor readings
-      zed_native.get_stereo_img(&img_left, &img_right);
-      l515.get_rgbd_frame(&img_rgb, &img_depth);
-      const auto timestamp = get_timestamp<std::chrono::microseconds>();
+      const int64_t timestamp = zed_native.get_stereo_img(&img_left, &img_right);
       // visual slam
-      const auto m = SLAM->feed_stereo_frame(img_left, img_right, timestamp / 1e6);
+      const pose_valid_tuple m = SLAM->feed_stereo_images_w_feedback(img_left, img_right, timestamp / 1e3);
```

Comment:
Why not directly reading out the `microsecond` clock?

Reply:
The internal timestamp maintained by OpenVSlam is microseconds. So this is for consistency.